src/twisted/newsfragments/10249.feature

```diff
@@ -0,0 +1,5 @@
+The ``all_non_platform`` extra has been renamed to ``all``,
+with the former name being a compatibility alias.  Platform-specific extras
+such as ``macos_platform`` and ``windows_platform`` have also been deprecated;
```

Comment:
Adding platform markers is good, but I don't that we should eliminate the platform integration extras as separate things. A user shouldn'thave to opt in to *every* Twisted dependency just to get platform integration.

Reply:
I don't disagree, just that currently that's how they're defined:

https://github.com/twisted/twisted/blob/5568fa24d60b470658b6cfc48be338f06cce57f0/setup.cfg#L96-L112

I don't have any problem defining e.g. `objc` and `win32` extras for just the `pyobjc-*` and `pywin32` packages respectively though, but someone will need to tell me to do it because I use neither Windows, macOS nor Twisted personally so I am unaware of how it is usually used.

Reply:
Oh, that's super weird. In that case your changes don't seem to represent a compatibility break so 👍